docs/Incompatibilities.md

```diff
@@ -0,0 +1,129 @@
+Incompatibilities between SGX-LKL and x86-64 Linux
+==================================================
+
+SGX-LKL is a port of Linux to a paravirtualised SGX environment, using LKL as a porting layer.
+SGX-LKL runs x86-64 programs and these programs execute the same machine instructions as they would outside of an enclave.
+The kernel; however, runs in quite a different environment to ring 0 on an x86-64 CPU and this can lead to some important differences.
+
+Compatibility differences
+-------------------------
+
+Some features are not supported at all and so applications that depend on them will not work.
+
+### Single process
+
+The SGX-LKL environment provides a single process.
+Any application that depends on spawning child processes will not work.
+
+In some future version, we expect to be able to support multiple processes but there will be no isolation between them.
+This means that using multiple processes for privilege separation will not be possible.
+
+### No MMU support
+
+A conventional x86-64 system exposes multiple privilege domains ('rings') and provides a memory management unit that allows each userspace process to have a separate virtual address space.
+This includes the ability to alias pages such that the same physical page is in several different locations in the virtual address space.
+In contrast, SGX exposes a single virtual address space and does not allow page mappings to be modified.
+
+At the system call layer, this means that there are some restrictions on `mmap`:
+
+ - Mapping anonymous memory works reliably.
+ - Shared mappings are not automatically written back.
+ - `MAP_FIXED` mappings should only be done over existing mappings: the kernel and userspace share an address space.
+   It is currently possible to do `MAP_FIXED` over kernel mappings, this will be fixed in a future version.
+
+These restrictions are close to those of uCLinux and SGX-LKL will eventually use the no-MMU code from Linux.
+
+Because SGX does not grant control over the MMU, it is impossible on SGX1 hardware to securely change page permissions.
+SGX-LKL delegates this to the untrusted host.
+As a result, any `mprotect` call, or any `mmap` with permissions weaker than read-write-execute, should be considered advisory.
+These calls will work correctly *in the absence of an attacker* but a malicious host can simply ignore these calls.
+This will be addressed on hardware with Enclave Dynamic Memory Management (EDMM) support.
+
+### Cooperative threading
```

Comment:
See single process comment.

Perhaps a small rewrite of the paragraph that leads into this "list" would be better.